src/api.py

```diff
@@ -0,0 +1,18 @@
+import requests 
+import hashlib
+def fetch(url):
+    con = requests.head(url)
+    r = requests.get(url)
+    name = url.split('/')[-1]
+    filename = '../download/'+name
+    with open(filename,'wb') as f:
+        f.write(r.content)
+    response = {
+        'filename' : filename,
+        'type_of_data' : con.headers['content-type'],
+        'size' : con.headers['content-length'],
+        'url' : url,
+        'checksum' : hashlib.md5(open(filename,'rb').read()).hexdigest()
```

Comment:
what if this fails? what if the file is not closed? And if the content is already in memory, why re-reading from disk? 
Also why privilege `md5` and make it the one value for `checksum`?

Reply:
Okay please can you suggest me some alternative for this